importers/algod/algod_importer.go

```diff
@@ -0,0 +1,81 @@
+package algodimporter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/algorand/go-algorand-sdk/client/v2/algod"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/rpcs"
+	"github.com/algorand/indexer/exporters"
+	"github.com/algorand/indexer/importers"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+)
+
+const importerName = "algod"
+
+type algodImporter struct {
+	aclient *algod.Client
+	logger  *logrus.Logger
+	cfg     ImporterConfig
+}
+
+var algodImporterMetadata = importers.ImporterMetadata{
+	ImpName:        importerName,
+	ImpDescription: "Importer for fetching block from algod rest endpoint.",
+	ImpDeprecated:  false,
+}
+
+// New initializes an algod importer
+func New() importers.Importer {
+	return &algodImporter{}
+}
+
+func (bot *algodImporter) Metadata() importers.ImporterMetadata {
+	return algodImporterMetadata
+}
+
+func (bot *algodImporter) Init(cfg string, logger *logrus.Logger) (err error) {
+	bot.logger = logger
+	if err := bot.unmarhshalConfig(cfg); err != nil {
+		return fmt.Errorf("connect failure in unmarshalConfig: %v", err)
+	}
+	var client *algod.Client
+	if !strings.HasPrefix(bot.cfg.NetAddr, "http") {
```

Comment:
nit: maybe use url.Parse() and check the Scheme? the current check doesn't cover a case like "httpsomeaddress".